Helper/Order.php

```diff
@@ -796,77 +812,132 @@ private function checkTotalsMismatch($order, $transaction)
         $this->bugsnag->registerCallback(function ($report) use ($transaction, $cart) {
             $report->setMetaData([
                 'TOTALS_MISMATCH' => [
-                    'Bolt' => $transaction->order->cart,
-                    'Store' => $cart,
+                    'Bolt Total' => $boltTotal,
+                    'Store Total' => $storeTotal,
+                    'Bolt Cart' => $transaction->order->cart,
+                    'Store Cart' => $cart
                 ]
             ]);
         });
-        $this->bugsnag->notifyError('Order Totals Mismatch', $comment);
 
-        // Save the order status and comment
         $order->save();
+
+        return true;
     }
 
     /**
      * Get (informal) transaction status to be stored with status history comment
      *
      * @param string $transactionState
-     * @param bool $fromHook
      * @return string
      */
-    private function getBoltTransactionStatus($transactionState, $fromHook)
+    private function getBoltTransactionStatus($transactionState)
     {
-
         return [
             self::TS_ZERO_AMOUNT => 'ZERO AMOUNT COMPLETED',
             self::TS_PENDING => 'UNDER REVIEW',
             self::TS_AUTHORIZED => 'AUTHORIZED',
+            self::TS_CAPTURED => 'CAPTURED',
             self::TS_COMPLETED => 'COMPLETED',
             self::TS_CANCELED => 'CANCELED',
             self::TS_REJECTED_REVERSIBLE => 'REVERSIBLE REJECTED',
             self::TS_REJECTED_IRREVERSIBLE => 'IRREVERSIBLE REJECTED',
-            self::TS_CREDIT_COMPLETED => $fromHook ? 'REFUNDED UNSYNCHRONISED' : 'REFUNDED'
+            self::TS_CREDIT_COMPLETED => Hook::$fromBolt ? 'REFUNDED UNSYNCHRONISED' : 'REFUNDED'
         ][$transactionState];
     }
 
+    /**
+     * Generate data to be stored with the transaction
+     *
+     * @param OrderModel $order
+     * @param \stdClass $transaction
+     * @param null|int $amount
+     */
+    private function formatTransactionData($order, $transaction, $amount)
+    {
+        return [
+            'Time' => $this->timezone->formatDateTime(
+                date('Y-m-d H:i:s', $transaction->date / 1000),
+                2,
+                2
+            ),
+            'Reference' => $transaction->reference,
+            'Amount'    => $order->getBaseCurrency()->formatTxt($amount / 100),
+            'Real ID'   => $transaction->id,
+        ];
+    }
+
+    /**
+     * Return the first unprocessed capture from the captures array (or null)
+     *
+     * @param OrderPaymentInterface $payment
+     * @param \stdClass $transaction
+     * @return mixed
+     */
+    private function getNewCapture($payment, $transaction)
```

Comment:
`getUnprocessedCapture`?  I don't think it's not necessarily chronological order

Reply:
@daisy1754
Renamed to getUnprocessedCapture. It seems the captures are stored in chronological order, but it does not matter since the record is kept of previously processed ones to avoid duplicates.